syntax = "proto3";

option go_package = ".;proto";

import "google/protobuf/empty.proto";

service User {
    rpc CheckPassword(CheckPasswordRequest) returns (CheckPasswordResponse);
    rpc GetUserList(PageRequest) returns (UserListResponse);
    rpc GetUserInfo(MobileRequest) returns (UserInfoResponse);
    rpc GetUserById(UidRequest) returns (UserInfoResponse);
    rpc UpdateUserInfo(UpdateUserRequest) returns (google.protobuf.Empty);
    rpc CreateUser(CreateUserRequest) returns (UserInfoResponse);
}

message PageRequest {
    sint32 page = 1;
    sint32 limit = 2;
}

message CheckPasswordRequest {
    string password = 1;
    string encryptedPassword = 2;
}

message CheckPasswordResponse {
    bool success = 1;
}

message UserListResponse {
    sint32 total = 1;
    repeated UserInfoResponse data = 2;
}

message MobileRequest {
    string mobile = 1;
}

message UserInfoResponse {
    sint32 id = 1;
    string mobile = 2;
    string nick_name = 3;
    string password = 4;
    string gender = 5;
    sint32 birthday =6;
    sint32 role = 7;
    string avatar = 8;
    string desc = 9;
    string country = 10;
    string provice = 11;
    string city = 12;
    string area = 13;
    string address = 14;
}

message UidRequest {
    sint32 uid = 1;
}

message CreateUserRequest {
    string mobile = 1;
    string nick_name = 2;
    string password = 3;
    string repassword = 4;
    string gender = 5;
    sint32 birthday =6;
    sint32 role = 7;
    string avatar = 8;
    string desc = 9;
    string country = 10;
    string provice = 11;
    string city = 12;
    string area = 13;
    string address = 14;
}

message UpdateUserRequest {
    sint32 uid = 1;
    CreateUserRequest data = 2;
}